### DISCUSSION TITLE
Artificial Intelligence (AI)

### DISCUSSION THREAD

**Discussion by rachaelbradley** [w3c/wcag3#38] [General]
As a group we need to get better informed about AI.  This discussion topic is a starting point. We'd like to bring in speakers when appropriate. 

On this discussion thread, please share resources and experts that the group should know about.

**Comment by ChrisLoiselle** [comment]
@rachaelbradley please keep me informed. Thanks!

**Reply by nayanecom** [reply]
Has the group reached out the stakeholders including those at OpenAI and others?

Here is the latest iteration of ChatGPT 4o - worth a watch for those who haven't seen it yet. https://www.youtube.com/live/DQacCB9tDaw?si=Ell_KIARyJSB_bRZ 

There is so much promise in what AI can do and is already doing and I am all for solutions that work at scale at the user and browser level.

**Comment by patrickhlauke** [comment]
Probably more as an aside, I've now heard at least twice in conversations about new success criteria something to the effect of "We won't have to worry about this anymore, as AI will do this for users automagically". I'd caution that AI (as in machine learning / ML) will still need actual correct reference material that it's trained on in order to attempt to magically correct content into the right shape (e.g. turning content into actual headings when it's not marked up as such), and for *that* to happen, there will still need to be guidelines/criteria that define what the correct content actually needs to look like so that appropriate training data can be created for these learning models (because if we just rely on ML trained on the state of the wild web as it is today ... we'll just end up with inaccessible output that just mimics the inaccessible state of the web today)

**Reply by patrickhlauke** [reply]
or, shorter: yes, AI/ML may be able to "correct" content, but only if we have a clear definition of what "correct" actually looks like, so we're not off the hook in terms of defining requirements/criteria

**Comment by lauracarlson** [comment]
Artificial Intelligence (AI) #38

https://github.com/w3c/wcag3/discussions/38#discussioncomment-7579238

Some Resources:

- [First, Do No Harm (Video)](https://www.youtube.com/watch?v=cRdfU-H0o6I) by Jutta Treviranus. "Artificial intelligence systems can make many decisions more efficiently, accurately, and consistently than humans. Their ability to learn human languages and perceptions and translate from one language or sensory mode to another is improving exponentially. Emerging large language models are threatening to pass the Turing test. However, these systems are also more efficient, accurate and consistent in enacting models of discrimination in the data. This amplifies, accelerates and automates the biases of the past. People with disabilities experience the extremes of both the potential and the risks. How can we harness the benefits without unleashing the harms?"
- [From Inclusion to Empowerment: The Role of AI in the Future of Accessibility and Inclusion (Video)](https://www.youtube.com/watch?v=uskt9v2DbtM) by Jutta Treviranus. "Kelsey Ruger and Jutta Treviranus explore the role of AI and Machine Learning in the future of accessibility, focusing on both the opportunities and challenges presented by these technologies. A moderator will engage these thought leaders in a dynamic conversation, sharing their expertise and insights on harnessing the power of AI while addressing potential risks and barriers…"
- [Preparing Accessibility for the Future and the Present (Video)](https://www.youtube.com/watch?v=xTWlRLG48eE) by Jutta Treviranus. "…Is it time to refocus and rethink our strategies? What is the role of accessibility in escalating deployment of AI and the emergence of new virtual worlds? How do we address the emerging threats and opportunities, including the risks in an accessibility dogma and the opportunity to practice what we preach?…" [Jutta's slides (PDF)](https://www.deque.com/axe-con/wp-content/uploads/2023/01/Juttas-Axe-con-Presentation.pdf).
- [Inclusive Design: Unlearning to Include and Innovate (Video) ](https://www.youtube.com/watch?v=uE543v8kj50)by Jutta Treviranus. "Jutta will present two assumptions that have become so central to our daily practice that we are no longer conscious of them. These two assumptions - tied to research methods, design, data analytics, AI and even our world views - have scaled and infused all of our thinking and practices, especially our foundational ideas about design, innovation and business. Every dilemma we are facing today can be traced back to these assumptions. She proposes turning these assumptions upside-down…"
- [Stay Human](https://buttondown.email/practicaltips/archive/stay-human/) by Matt May. "Why AI/ML harm is a pressing equity issue…The AI/ML space is not being driven by humanists. Far from it. In fact, those who have documented actual harms with existing AI models have been fired from companies developing them…"
- [Artificial Intelligence's Utility for Accessibility](https://dl.acm.org/doi/fullHtml/10.1145/3597638.3614548) By Kate S Glazko, Momona Yamagami, Aashaka Desai, Kelly Avery Mack, Venkatesh Potluri, Xuhai Xu, Jennifer Mankoff.. "…Our work demonstrates the potential for GAI today to be used by people with disabilities to provide on-demand support for their accessibility needs, in low-stakes, easily verifiable contexts. Unsurprisingly, some of the limitations we encountered reflect GAI's current ability to parrot [5] information without truly integrating it. This can explain why it struggled, for example, to translate accessibility guidelines into documents consistently… our work also uncovered problems that require directed ongoing attention to fix, such as ableist errors; accessible and complementary approaches to verification; and better training data. The subtleties of some of these errors raise questions about how and whether it is possible to address, or even detect them automatically…"
- [Why Doesn't AI Work for Producing Accessible Code?](https://alastairc.uk/2023/11/why-doesnt-ai-work-for-producing-accessible-code/) By Alastair Campbell. @alastc  "The short answer to the title is: Accessibility is not an average. For a longer answer, read on…"
- [Image Describer (Chrome extension)](https://chromewebstore.google.com/detail/ogoddjgogmlndofcpkljmmdobjpfdolf) By Cameron Cundiff. @ckundo "Describe images in a web page using the context menu for accessibility for people who are blind or who have low vision…"

**Reply by u9000-Nine** [reply]
Definite +1 on Alastair's piece. I read that today and was going to add it, but I'm glad to see it already has been.

**Comment by GrahamTheDevRel** [comment]
A few thoughts on AI (random thoughts, just to spark discussion):

## 1. AI generated content (dynamic)
Any AI generated content that is generated "on the fly" is likely to be impossible to test in any meaningful way. For example, the content may pass a test 9/10 times but then "hallucinate" (offer a completely incorrect and even non-sensical and unrelated answer) on the 10th generation and create something completely inaccessible. 

Note that "hallucinations" can happen even if the input is identical each time due to the nature of how AI models work. 

As such dynamically generated AI content will need some "average score" or "number of tests passed" or similar criterion, or be excluded entirely. 

**note:** for clarity when I say dynamically, I mean on each page load or each action AI generated content is produced, for example an AI powered chat application that uses formatting, in order to test that formatting is valid is nearly impossible and at best could be performed as an average using automated tooling. An additional thought here is as an external tester, this is especially difficult to test as it would rely on having the appropriate tooling, or being provided with the data for 10s or 100s of examples and being able to analyse that data.

## 2. AI Generated content (static)
Any content that is created using AI, but then only published once (i.e. a blog post is "published once" vs a response from an AI agent which may change each time and fall under the previous heading) should still fall under WCAG. I see no reason why it should be excluded, even if it is auto published. Maybe there is space for a guideline / note that all AI generated "static / single publication / whatever we call it" content is subject to the same Guidelines for clarity.

## 3. AI interfaces / Front-ends / GUIs
These should be fully testable and conform to WCAG etc. There should be no limitations here.

## 4. Identification of AI Agents
Consistent and repeated identification of AI generated content is essential, especially in the context of chat bots.

A lot of people (especially those who are more vulnerable) can become attached and involved with AI generated personalities, conversations etc. 

Sufficient safe guards, warnings and reminders will likely be a necessary part of WCAG or similar guidance. 

Example: Replika is an AI "partner" that you can converse with. People who had trauma and did not feel they could speak to a person about it used the chatbot to talk about their trauma. Then they updated the model that powers the chat to be more "advertiser and investor friendly" and a lot of conversations were not possible as they contained content that was graphic, Not Safe For Work (NSFW) etc. This caused a lot of pain and distress for those users who were relying on this chat bot to process things.

While the ethics of updating models etc. is outside the scope of WCAG, perhaps identification and reminder of AI agents, plus sufficient notice of potential changes to the model etc. *may* be applicable to protect against scenarios like this.

## 5. Protection from Bias
AI is biased. It is only as good as the data it is fed, and our data is biased also (especially towards "western culture" for things like OpenAI / ChatGPT.)/
 
People would benefit from being pre-warned about biases and reminded about them regularly. Additionally there are a lot of scenarios where AI may not have been fed information about disabled user's needs etc.

Additionally bias against disabled people is likely to exist.

Example: An AI system that uses a webcam and allows people to virtually try on clothes. This system may not have been trained on people using wheelchairs and therefore may not recognise them, making the system unusable. 

Example: An AI agent is used to pre-screen job applications. This may result in biases due to "unusual" working patterns that someone with a disability may have had. If this is then used to make decisions about applicants as part of a decision process then it may add to exclusionary practices.

**Note:** yet again I have no idea if this is under the purview of WCAG, just another random thought to spark discussion.

These are just my initial ramblings / thoughts, I will have a think and may write a more robust and more well though out article / response here.

**Reply by u9000-Nine** [reply]
> 1. AI generated content (dynamic)

I think these are definitely things we need to consider when writing the outcomes, but I disagree that they should be exempt simply because they're hard to evaluate. Worst case, we rely on assertions about the training data.

Under the old model, I can see an argument for treating them as authoring tools (thus under the Authoring Tools Accessibility Guidelines instead of the Web Content Accessibility Guidelines), but not for excluding them entirely. I see no reason why dynamically generated AI content should be excluded under the new model, though.

> While the ethics of updating models etc. is outside the scope of WCAG

I think this is covered under Prevent Harm's "[Don't Deceive Users](https://docs.google.com/document/d/1S9K4i0CAiTHCWXD0_LnpS90E2l9VC_Xs18UmOXhGzaw/edit?pli=1#heading=h.mzmzfctjj095)" and "[Protect from Harm from Algorithms](https://docs.google.com/document/d/1S9K4i0CAiTHCWXD0_LnpS90E2l9VC_Xs18UmOXhGzaw/edit?pli=1#heading=h.a0f2o2kireqg)".
